src/user/proposals/UserProposalDashboard/DashboardRightPanel/Comments/Comments.js

```diff
@@ -2,71 +2,143 @@ import React, { Component } from "react";
 import "./Comments.scss";
 import { ListGroup, Image } from "react-bootstrap";
 import userIcon2 from "../../../../../images/userIcon2.jpg";
+import socket from "../../../../dashboard/utils/socket";
+import { connect } from "react-redux";
+import { getUserProposalNotifications } from "../../../../../actions/proposalActions";
+import { getProposalNotifications } from "../../../../../actions/notificationAction";
 
 class Comments extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      socket: socket,
+      notifications: [],
+      commentNotifications: [],
+    };
   }
+
+  componentDidMount() {
+    const data = {
+      userId: localStorage.getItem("userId"),
+    };
+
+    this.props.getProposalNotifications();
+    this.props.getUserProposalNotifications(data);
+
+    this.state.socket.on("new proposal created", (data) => {
+      data.newNotification = true;
+      data.createdAt = new Date().toString().substring(0, 24);
+      this.setState({
```

Comment:
@AuraOfDivinity Instead of repeating this every time, just make a function and pass the value to set notifications